pkg/odo/cli/delete/namespace/namespace.go

```diff
@@ -78,6 +78,15 @@ func (do *DeleteOptions) Validate(ctx context.Context) (err error) {
 
 // Run contains the logic for the 'odo delete namespace' command
 func (do *DeleteOptions) Run(ctx context.Context) error {
+	log.Infof("Searching %s to delete, please wait...", do.commandName)
+	exists, err := do.clientset.ProjectClient.Exists(do.namespaceName)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		log.Errorf("No %s named %q found", do.commandName, do.namespaceName)
+		return nil
```

Comment:
I think we should continue to return an error, like so (you can also remove the `log.Errorf` line above to avoid redundant messages):

```suggestion
		return fmt.Errorf("no %s named %q found", do.commandName, do.namespaceName)
```

Otherwise, the command would be considered as succeeding, which was not the case before. 
Doing so should also fix the integration tests which are not passing on this PR ([example](https://github.com/redhat-developer/odo/pull/6233#issuecomment-1283391935)).